### ISSUE TITLE
libarchive.so.14: cannot open shared object file

### ISSUE THREAD

**Issue by TheAssassin** [AppImage/AppImageKit#476]
```
./appimaged-x86_64.AppImage 
appimaged: error while loading shared libraries: libarchive.so.14: cannot open shared object file: No such file or directory
```

On Ubuntu 16.04, apparently appimaged won't run due to libarchive 1.4 not being shipped. This is related to #343, where @ShalokShalom reported a system was missing libarchve 1.3. The fun fact is that my system ships with libarchive 1.3 (therefore appimaged used to work a few months ago).

I think we really have to either `dlopen()` libarchive, or just ship our own (really not a problem for appimaged in my opinion). Or, as libarchive is really a legacy dependency to support type 1 AppImages, follow @probono's suggestion in #353. I'd ship the `.so` for now as a quick fix, and implement #353 as soon as possible.

**Comment by simoniz0r** [comment]
I am having the same issue.

**Comment by probonopd** [comment]
Possibly caused by https://github.com/libarchive/libarchive/issues/817. We may well want to dump libarchive altogether since the upstream project doesn't seem to be willing to fix this issue.

**Comment by drohm** [comment]
Is there any potential temporary workaround atm?

**Comment by simoniz0r** [comment]
Is there something stopping you from just putting libarchive in the AppImage or are you already doing this?

**Comment by drohm** [comment]
@simoniz0r I'm not actually building an AppImage, I'm wanting to use appimaged to watche locations like ~/bin and ~/Downloads for AppImages and if it detects some, registers them with the system, so that they show up in the menu, have their icons show up, MIME types associated, etc.  Am I understaning the tool incorrectly?

**Comment by simoniz0r** [comment]
@drohm Sorry, that was directed at the devs, not you lol

**Comment by probonopd** [comment]
> I'm wanting to use appimaged to watche locations like ~/bin and ~/Downloads for AppImages and if it detects some, registers them with the system, so that they show up in the menu, have their icons show up, MIME types associated, etc.

Correct, that's what it's supposed to do. But apparently libarchive in Ubuntu 16.04 broke due to them changing the version number for no clear reason.

**Comment by simoniz0r** [comment]
What is preventing you from putting libarchive in the AppImage @probonopd ?

**Comment by probonopd** [comment]
Adding size and "bloat" just because libarchive and the distributions can't seem to agree on a version number seems odd to me. That being said, see #355.

**Comment by TheAssassin** [comment]
@probonopd we could just drop support for type 1 AppImages as well. If it is such an effort to retain support, why bother the effort? Type 1 is deprecated, and developers should switch. I'd just show an error message instead of bothering to support them.

**Comment by ShalokShalom** [comment]
Agree +1

**Comment by simoniz0r** [comment]
@probonopd You're really concerned about ~200KBs?  If I were you, I'd be more concerned about people like myself who have been trying to use appimaged for about a month now, but can't due to you not including libarchive.

**Comment by TheAssassin** [comment]
@simoniz0r please check whether the version built in #355's Travis build (there's a transfer.sh link somewhere at the end of the log) works for you. If it does, I'll happily merge @darealshinji's work.

**Comment by simoniz0r** [comment]
@TheAssassin Yep, that build works!

![image](https://user-images.githubusercontent.com/16279904/31152461-d590868e-a861-11e7-824d-cd5bc7e9ad76.png)

**Comment by TheAssassin** [comment]
Merged #355.
